### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by jakobwenzel** [andsens/homeshick#132]
I use a mix of Linux and Windows machines. On Windows, I use MSys to get a useable shell.

It would be great if homeshick could work in this environment. Apart from failing to parse the git version string (it's "1.9.4.msysgit.2"), the only thing that I see that would need changing is symbolic link creation. Since Vista, Windows supports symlinks, so it's possible: http://stackoverflow.com/questions/18641864/git-bash-shell-fails-to-create-symbolic-links#25394801

**Comment by andsens** [comment]
I agree that Windows support would be quite nice, but symlinking is not the only requirement. We also need a way to replace `readlink` and other various tools.  
I'd be open to merging a PR that passes all the unit tests, but I myself will have a hard time maintaining it, since I do not work with Windows at all.

**Comment by andsens** [comment]
Closing due to lack of activity (I assume this means lack of interest as well).

Feel free to re-open if you have a working version. Depending on the extent of the modifications I might merge, otherwise it will definitely get a mention in the wiki and the README.

**Comment by KitKat31337** [comment]
@andsens I am porting this into Powershell to use on my windows hosts. https://stash.firekitten.net/Firekitten/homepsick 

I am targeting powershell 7 which can be run on other platforms, and doing my best to keep it platform agnostic.  But its purpose is definitely to allow someone to use it along side homeshick with their windows hosts using the same repos and methodology.

If you are interested in discussing at all or having any input, please let me know.

**Comment by andsens** [comment]
What a brilliant effort @KitKat31337 :-D  
Something like #77 would definitely be helpful in this case. The current version of homeshick is kind of "done". Meaning I still fix bugs and merge some minor features, but nothing major will change. At some point I might do a refactor and do the argument parsing with [docopt.sh](https://github.com/andsens/docopt.sh), but that's quite a bit in the future. Unfortunately it doesn't seem like anybody ported docopt to PS :-/

Once you think your port is done enough, let me know and I will link to it from the README. And don't hesitate to ask if I can help in any way (that doesn't involve coding PS (I suck at that :-) ). Maybe open a new issue to discuss everything.

**Comment by KitKat31337** [comment]
@andsens 
Strictly speaking its functional-ish already, but requires you to link all the functions manually as I have not made a module manifest yet.  I have the ability to clone a castle, and to link a castle.  Next up I will have the ability to pull from git to update castle, which is the basic bare bones MVP (at least for me).  At that time I plan to make the module manifest so you can do a one line import and use all the functions.  I will look at #77 more, just OS Filtering seems pretty simple.

**Comment by KitKat31337** [comment]
@andsens I moved the project over to github to make it simpler for people to colloaborate if they wanted to fork and PR.

https://github.com/KitKat31337/homepsick
